src/nbl/video/CVulkanDescriptorPool.cpp

```diff
@@ -26,45 +26,46 @@ void CVulkanDescriptorPool::setObjectDebugName(const char* label) const
 	vkSetDebugUtilsObjectNameEXT(vulkanDevice->getInternalObject(), &nameInfo);
 }
 
-core::smart_refctd_ptr<IGPUDescriptorSet> CVulkanDescriptorPool::createDescriptorSet_impl(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout)
+bool CVulkanDescriptorPool::createDescriptorSets_impl(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, SDescriptorOffsets* const offsets, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
 {
-    if (layout->getAPIType() != EAT_VULKAN)
-        return nullptr;
-
     VkDescriptorSetAllocateInfo vk_allocateInfo = { VK_STRUCTURE_TYPE_DESCRIPTOR_SET_ALLOCATE_INFO };
     vk_allocateInfo.pNext = nullptr; // pNext must be NULL or a pointer to a valid instance of VkDescriptorSetVariableDescriptorCountAllocateInfo
 
     vk_allocateInfo.descriptorPool = m_descriptorPool;
-    vk_allocateInfo.descriptorSetCount = 1u;
-
-    VkDescriptorSetLayout vk_dsLayout = IBackendObject::device_compatibility_cast<const CVulkanDescriptorSetLayout*>(layout.get(), getOriginDevice())->getInternalObject();
-    vk_allocateInfo.pSetLayouts = &vk_dsLayout;
+    vk_allocateInfo.descriptorSetCount = count;
 
-    VkDescriptorSet vk_descriptorSet;
+    core::vector<VkDescriptorSetLayout> vk_dsLayouts(count);
+    for (uint32_t i = 0; i < count; ++i)
+    {
+        if (layouts[i]->getAPIType() != EAT_VULKAN)
+            return false;
```

Comment:
because `IDescriptorPool` already checks for compat before calling this function, you can just plainly assert

Reply:
Resolved.